docs/products/valkey/howto/migrate-caching-valkey-to-aiven-for-valkey.md

```diff
@@ -22,71 +24,158 @@ Before starting the migration process, ensure the following:
+      - **Keep replicating**: If continuous data synchronization is needed. Avoid system
+        updates or configuration changes during active replication to  prevent unintended
+        migrations.
 
-:::note
-When replication is active, Aiven for Valkey ensures your data stays in sync by
-continuously synchronizing new writes from the source database.
+      :::note
+      When replication is active, Aiven for Valkey ensures your data stays in sync by
+      continuously synchronizing new writes from the source database.
+      :::
+
+</TabItem>
+<TabItem value="k8" label="Aiven Operator for Kubernetes®">
+:::tip
+See
+[Aiven Operator for Kubernetes configuration options for Valkey](https://aiven.github.io/aiven-operator/api-reference/valkey.html).
+:::
+
+1. Use the Aiven API to change the type of your service from `redis` to `valkey`: Call the
+   [ServiceServiceTypeUpdate](https://api.aiven.io/doc/#tag/Service/operation/ServiceServiceTypeUpdate)
+   API endpoint, replacing `PROJECT_NAME` and `SERVICE_NAME` with meaningful values.
+
+   ```bash {4}
+   curl -X PATCH "https://api.aiven.io/v1/project/PROJECT_NAME/service/SERVICE_NAME/service_type" \
+   -H "Authorization: Bearer $AIVEN_TOKEN" \
+   -H "Content-Type: application/json" \
+   -d '{"service_type": "valkey"}'
+   ```
+
+1. [Get authenticated and authorized](https://aiven.github.io/aiven-operator/authentication.html)
+   to use the Aiven Operator for Kubernetes.
+1. In your service manifest file, change `kind: Redis` to `kind: Valkey`:
+
+   ```yaml {3}
+   # sample_redis.yaml
+   apiVersion: aiven.io/v1alpha1
+   kind: Valkey
+   metadata:
+     name: METADATA_NAME
+   spec:
+     authSecretRef:
+       name: aiven-token
+       key: token
+
+     connInfoSecretTarget:
+       name: SECRET_NAME
+
+     project: PROJECT_NAME
+
+     cloudName: CLOUD_AND_REGION_NAME
+     plan: SERVICE_PLAN_NAME
+
+     maintenanceWindowDow: DAY_OF_WEEK
+     maintenanceWindowTime: HH:MM:SS
+
+     technicalEmails:
+       - email: test@example.com
+
+     connInfoSecretTargetDisabled: false
+   ```
+
+1. Update the service by applying the configuration:
+
+   ```bash
+   kubectl apply -f sample_redis.yaml
+   ```
+
+   This creates an Aiven for Valkey resource: a service managed by the Aiven Operator for
+   Kubernetes.
+
+1. In the Aiven for Redis manifest file, add `controllers.aiven.io/deletion-policy: Orphan`
+   under `metadata` > `annotations`.
+
+1. Delete the Aiven for Redis resource.
+
+   As a result, the Aiven for Redis resource is removed and the new Aiven for Valkey resource
+   persists.
+
+:::important[handling secrets]
+
+- After the migration, the connection secret of your new Aiven for Valkey service has
+  prefixes `REDIS`. To update the secret, change the prefixes manually to `VALKEY`.
+- If you set `connInfoSecretTargetDisabled` to `false` in the service manifest, no secret
+  is created.
+- The secret name is based on `connInfoSecretTarget`.
```

Comment:
@byashimov How would you word this? I think we may not want to suggest changing the prefixes manually to VALKEY? Just noting that we don't touch the connection info secret during this upgrade might be enough?

Reply:
I believe when your remove the Redis, the secret disappears and VK doesn't have one.

Another problem here is that there must be running applications. I would go for:

1. Remove deletion annotation (or mark Orphan) the Redis instance. The secret contains `REDIS_` keys and has the service name.
2. Create a VK instance with a new `connInfoSecretTarget.name`. The secret contains `VALKEY_` keys _and_ it has a **new** name.
3. Update your applications, point them to the new secret name. The apps must be either using the new prefix, or it must be set to `REDIS`.
4. Make sure nothing depends on the Redis secret.
5. Delete redis

If `connInfoSecretTargetDisabled` is set to `false` on Redis, hence the apps have a different source of credentials, steps 3 and 4 can be skipped.

Reply:
This would be disruptive to the user as they'd be creating a new Valkey service instead of migrating the old one with [ServiceServiceTypeUpdate](https://api.aiven.io/doc/#tag/Service/operation/ServiceServiceTypeUpdate).

I believe this is not acceptable to the customer.

Reply:
Ideally, we'd like to b able to set up replication automatically then have our users update their apps as they please and then delete the redis instances. We have users with redis-as-cache and redis-as-datastore and both migration stories would be well covered with replication + switch over and delete.